packages/web3-common/src/types.ts

```diff
@@ -35,3 +42,151 @@ export interface JsonRpcPayload<T = unknown[]> extends JsonRpcRequest<T> {
+	readonly id?: string;
+	readonly blockNumber?: HexString;
+	readonly transactionIndex?: HexString;
+	readonly address?: HexString;
+}
+export interface LogsOutput {
+	readonly id?: string;
+	readonly blockHash?: HexString;
+	readonly transactionHash?: HexString;
+	readonly logIndex?: bigint | number;
+	readonly blockNumber?: bigint | number;
+	readonly transactionIndex?: bigint | number;
+	readonly address?: string;
+}
+
+export interface BlockInput {
+	readonly gasLimit: HexString;
+	readonly gasUsed: HexString;
+	readonly size: HexString;
+	readonly timestamp: HexString;
+	readonly number?: HexString;
+	readonly difficulty?: HexString;
+	readonly totalDifficulty?: HexString;
+	readonly transactions?: TransactionInput[];
+	readonly miner?: HexString;
+	readonly baseFeePerGas?: HexString;
+}
+
+export interface BlockOutput {
+	readonly gasLimit: bigint | number;
+	readonly gasUsed: bigint | number;
+	readonly size: bigint | number;
+	readonly timestamp: bigint | number;
+	readonly number?: bigint | number;
+	readonly difficulty?: bigint | number;
+	readonly totalDifficulty?: bigint | number;
+	readonly transactions?: TransactionOutput[];
+	readonly miner?: HexString;
+	readonly baseFeePerGas?: bigint | number;
+}
+
+export interface ReceiptInput {
+	readonly [x: string]: unknown;
+	readonly blockNumber?: HexString;
+	readonly transactionIndex?: HexString;
+	readonly cumulativeGasUsed: HexString;
+	readonly gasUsed: HexString;
+	readonly logs?: LogsInput[];
+	readonly contractAddress?: HexString;
+	readonly status?: string;
+}
+
+export interface ReceiptOutput {
+	readonly blockNumber?: bigint | number;
+	readonly transactionIndex?: bigint | number;
+	readonly cumulativeGasUsed: bigint | number;
+	readonly gasUsed: bigint | number;
+	readonly logs?: LogsOutput[];
+	readonly contractAddress?: HexString;
+	readonly status: boolean;
+}
+
+export interface PostInput {
+	readonly ttl?: HexString;
+	readonly workToProve?: HexString;
+	readonly priority?: HexString;
+	readonly expiry?: HexString;
+	readonly sent?: HexString;
+	readonly workProved?: HexString;
+	readonly topics?: HexString[];
+}
+
+export interface PostOutput {
+	readonly ttl?: bigint | number;
+	readonly workToProve?: bigint | number;
+	readonly priority?: bigint | number;
+	readonly expiry?: bigint | number;
+	readonly sent?: bigint | number;
+	readonly workProved?: bigint | number;
+	readonly topics?: string[];
+}
+
+export interface SyncInput {
+	readonly startingBlock: HexString;
+	readonly currentBlock: HexString;
+	readonly highestBlock: HexString;
+	readonly knownStates?: HexString;
+	readonly pulledStates?: HexString;
+}
+
+export interface SyncOutput {
+	readonly startingBlock: bigint | number;
+	readonly currentBlock: bigint | number;
+	readonly highestBlock: bigint | number;
+	readonly knownStates?: bigint | number;
+	readonly pulledStates?: bigint | number;
+}
+
+export enum PredefinedBlockNumbers {
+	LATEST = 'latest',
+	PENDING = 'pending',
+	EARLIEST = 'earliest',
```

Comment:
```suggestion
	latest = 'latest',
	pending = 'pending',
	earliest = 'earliest',
```

These need to be lower case to support indexing by enum value, and lowercase should be okay since [Lodestar is doing it](https://github.com/ChainSafe/lodestar/blob/456b50ee628eb60e3daba23cbb596cef41ea6113/packages/lodestar/src/sync/interface.ts#L68)

**NOTE** If you merge this suggestion, we need to update the use of the enums elsewhere

Reply:
Created #4462 to track this